### PR TITLE
docs(tagline): modify docs to verify component

### PR DESCRIPTION
**Pull request by billhimmelsbach**
I went through and [verified the Tagline component](https://github.com/cfpb/design-system-react/issues/205) as part of the work on the banner component. Only small changes to the documentation appear to be needed. I didn't add the deprecated "Extra large tagline" story from the design system, since it was deprecated and all, but we can chat about that at the DSR sync. 

## Changes

- add component description
- remove heading
- change stories to sentence case

## How to test this PR

1. Take a peek at the DSR and see how it looks

## Screenshots
Screenshot 2023-10-19 at 9 48 05 AM

## Notes
I didn't mark this component as "verified" in this PR, since I'm not sure if we have an official stamp of approval process yet, so I'll hold off till at least we chat about it at the sync.

### CODE REVIEW

**Comment by billhimmelsbach** [comment]
I didn't mark this as "verified" in this PR, since I'm not sure if we have an official stamp of approval process yet, so I'll hold off till at least we chat about it at the sync.

**Review by meissadia — APPROVED** [review]

**Pull request merged**
